mapie/metrics.py

```diff
@@ -737,3 +740,578 @@ def hsic(
     coef_hsic = np.sqrt(np.matrix.trace(hsic_mat, axis1=1, axis2=2))
 
     return coef_hsic
+
+
+def jitter(
+    x: NDArray,
+    noise_amplitude: float = 1e-8,
+    random_state: Optional[Union[int, np.random.RandomState]] = None
+) -> NDArray:
+    """
+    Add a tiny normal distributed perturbation to an array x.
+
+    Parameters
+    ----------
+    x : NDArray
+        The array to jitter.
+
+    noise_amplitude : float, optional
+        The tiny relative noise amplitude to add, by default 1e-8.
+
+    random_state: Optional[Union[int, RandomState]]
+        Pseudo random number generator state used for random sampling.
+        Pass an int for reproducible output across multiple function calls.
+
+    Returns
+    -------
+    NDArray
+        The array x jittered.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.metrics import jitter
+    >>> x = np.array([0, 1, 2, 3, 4])
+    >>> res = jitter(x, random_state=1)
+    >>> res
+    array([0.        , 0.99999999, 1.99999999, 2.99999997, 4.00000003])
+    """
+    n = len(x)
+    random_state_np = check_random_state(random_state)
+    noise = noise_amplitude * random_state_np.normal(size=n)
+    x_jittered = x * (1 + noise)
+    return x_jittered
+
+
+def sort_xy_by_y(x: NDArray, y: NDArray) -> Tuple[NDArray, NDArray]:
+    """
+    Sort two arrays x and y according to y values.
+
+    Parameters
+    ----------
+    x : NDArray of size (n_samples,)
+        The array to sort according to y.
+
+    y : NDArray of size (n_samples,)
+        The array to sort.
+
+    Returns
+    -------
+    Tuple[NDArray, NDArray]
+        Both arrays sorted.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.metrics import sort_xy_by_y
+    >>> x = np.array([1, 2, 3, 4, 5])
+    >>> y = np.array([5, 4, 3, 1, 2])
+    >>> x_sorted, y_sorted = sort_xy_by_y(x, y)
+    >>> print(x_sorted)
+    [4 5 3 2 1]
+    >>> print(y_sorted)
+    [1 2 3 4 5]
+    """
+    x = cast(NDArray, column_or_1d(x))
```

Comment:
Why do you cast to `NDArray` if that's already the type of the variable as an input?

Reply:
Good point ! Just copy-pasted from `hsic` function...
remove the cast : done